### PR TITLE
Add types and clean TYPEOF

### PR DESCRIPTION
**Pull request by ftence**
Also merged reducer, mapper and bucketizer under aggregator because they cannot be differentiated.

### CODE REVIEW

**Comment by randomboolean** [comment]
The function typeof(Class c) is needed by some other classes.

**Comment by ftence** [comment]
Which ones? Couldn't find any.

**Comment by randomboolean** [comment]
Mostly in extensions.

**Comment by ftence** [comment]
Can you be more precise?

**Comment by randomboolean** [comment]
Here for example: https://github.com/senx/warp10-ext-arrow/search?q=typeof&unscoped_q=typeof

**Comment by ftence** [comment]
Thanks for the heads-up @randomboolean, this is fixed.

File: warp10/src/main/java/io/warp10/script/functions/TYPEOF.java [hunk 1]
```diff
@@ -97,57 +108,72 @@ public static String typeof(Object o) {
...
     } else {
-      return (c.getSimpleName().toUpperCase());
+      return defaultType(c);
+    }
+  }
+
+  private static String defaultType(Class c) {
+    String canonicalName = c.getCanonicalName();
+    if (null == canonicalName) {
+      return "X-(Local/Anonymous Class)";
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/functions/TYPEOF.java]: It would be better to use neither parens nor spaces.

X-Local-Anonymous-Class?

**Pull request merged**